src/test/java/uk/gov/hmcts/darts/arm/service/CleanupArmResponseFilesServiceImplTest.java

```diff
@@ -0,0 +1,139 @@
+            externalObjectDirectoryRepository,
+            objectRecordStatusRepository,
+            externalLocationTypeRepository,
+            armDataManagementApi,
+            userIdentity,
+            armDataManagementConfiguration,
+            currentTimeHelper
+        );
+    }
+
+    @Test
+    void cleanupResponseFilesSuccess() {
+
+        when(externalLocationTypeRepository.getReferenceById(3)).thenReturn(externalLocationTypeArm);
+
+        when(objectRecordStatusRepository.findById(2)).thenReturn(Optional.of(objectRecordStatusStored));
+        when(objectRecordStatusRepository.findById(17)).thenReturn(Optional.of(objectRecordStatusArmResponseProcessingFailed));
+        when(objectRecordStatusRepository.findById(18)).thenReturn(Optional.of(objectRecordStatusArmResponseChecksumFailed));
+        when(objectRecordStatusRepository.findById(19)).thenReturn(Optional.of(objectRecordStatusArmResponseManifestFailed));
+
+        when(armDataManagementConfiguration.getResponseCleanupBufferDays()).thenReturn(0);
+
+        when(media.getId()).thenReturn(456);
+
+        when(externalObjectDirectory.getId()).thenReturn(123);
+        when(externalObjectDirectory.getMedia()).thenReturn(media);
+
+        OffsetDateTime testTime = OffsetDateTime.now().plusMinutes(10);
+        when(currentTimeHelper.currentOffsetDateTime()).thenReturn(testTime);
+
+        when(externalObjectDirectoryRepository.findByStatusInAndStorageLocationAndResponseCleanedAndLastModifiedDateTimeBefore(
+            any(),
+            any(),
+            anyBoolean(),
+            any()
+        )).thenReturn(List.of(externalObjectDirectory));
+
+        String inputUploadBlobFilename = "123_456_1_6a374f19a9ce7dc9cc480ea8d4eca0fb_1_iu.rsp";
+        List<String> inputUploadFilenameResponseBlobs = new ArrayList<>();
+        inputUploadFilenameResponseBlobs.add(inputUploadBlobFilename);
+        when(armDataManagementApi.listResponseBlobs("123_456_")).thenReturn(inputUploadFilenameResponseBlobs);
+
+        List<String> hashcodeResponseBlobs = new ArrayList<>();
+        String hashcode = "6a374f19a9ce7dc9cc480ea8d4eca0fb";
+        String createRecordFilename = "6a374f19a9ce7dc9cc480ea8d4eca0fb_a17b9015-e6ad-77c5-8d1e-13259aae1895_1_cr.rsp";
+        String uploadFileFilename = "6a374f19a9ce7dc9cc480ea8d4eca0fb_04e6bc3b-952a-79b6-8362-13259aae1895_1_uf.rsp";
+        hashcodeResponseBlobs.add(createRecordFilename);
+        hashcodeResponseBlobs.add(uploadFileFilename);
+        when(armDataManagementApi.listResponseBlobs(hashcode)).thenReturn(hashcodeResponseBlobs);
+
+        when(armDataManagementApi.deleteBlobData(any())).thenReturn(true);
+
+        when(userIdentity.getUserAccount()).thenReturn(testUser);
+
+        cleanupArmResponseFilesService.cleanupResponseFiles();
+
+        verify(externalObjectDirectoryRepository, times(1)).saveAndFlush(externalObjectDirectoryEntityCaptor.capture());
```

Comment:
No need for `times(1)` in verify - that is the default.
I guess if we didn't mock the externalObjectDirectory we could verify the flag field `responseCleaned`